Fixed cohorts section step 2

asadazam93 opened the pull request:
[TNL-8135](https://openedx.atlassian.net/browse/TNL-8135)

Updated the cohorts sections to match the Figma design


Expected design in Figma
<img width="423" alt="Screenshot 2021-04-16 at 4 07 08 PM" src="https://user-images.githubusercontent.com/10988308/115016626-bf029080-9ece-11eb-9cfb-6e2a30ee2927.png">




Outcome after change:
<img width="791" alt="Screenshot 2021-04-16 at 4 42 37 PM" src="https://user-images.githubusercontent.com/10988308/115019622-e22f3f00-9ed2-11eb-8dc6-cadbefdf82fb.png">

davidjoy commented on src/pages-and-resources/discussions/app-config-form/apps/legacy/LegacyConfigForm.jsx:
```diff
@@ -36,8 +37,8 @@ function LegacyConfigForm({
-        <AppConfigFormDivider />
-        <DivisionByGroupFields
+        <AppConfigFormDivider thick />
+        <InContextDiscussionFields
```
`InContextDiscussionFields` can't be here included in `LegacyConfigForm` - this is the legacy form and doesn't support in-context discussions.  I don't think the Figma files are correct in this specific case; I think they're showing the "new" discussions fields.

davidjoy commented on src/pages-and-resources/discussions/app-config-form/apps/shared/DivisionByGroupFields.jsx:
```diff
@@ -31,17 +32,7 @@ function DivisionByGroupFields({
-              className="ml-4"
-              id="allowDivisionByUnit"
-              checked={values.allowDivisionByUnit}
-              label={intl.formatMessage(messages.allowDivisionByUnitLabel)}
```
What happened to this field?  I think we need to pause and make sure we have the right fields for this form - I'm not sure the effect of just pulling this out and not updating the rest of the data model.

davidjoy reviewed (changes requested):
A few stop ship problems.  In-context discussion isn't supported by the legacy config form and can't be in there, and I also see a field that was removed without cleaning up any of its messages or the data model.  It's not clear to me that `allowDivisionByUnit` should be removed at all.

davidjoy commented:
(also, please use conventional commit messages in your PR title)

asadazam93 commented:
@davidjoy Okay I removed the in-context discussions. I was confused because it was visible in the designs and also there was another ticket to re order elements in the form and that also mentioned in-context discussions at the top. I kept the data model for “Allow cohort division for each course unit” in case it needs to be used in future, but I have removed that as well now

davidjoy commented on src/pages-and-resources/discussions/app-config-form/apps/shared/DivisionByGroupFields.jsx:
```diff
@@ -14,10 +14,11 @@ function DivisionByGroupFields({
   return (
     <>
-      <h5>{intl.formatMessage(messages.divisionByGroup)}</h5>
+      <h5 className="text-gray-500 mt-4">{intl.formatMessage(messages.divisionByGroup)}</h5>
```
Nit: in general, it's a good practice to try to consistently use bottom margins to space things out, rather than top margins.  This way each component is responsible for the space it wants below it.  It makes things a bit more predictable.  In this component for instance, I'd have this `h5` define `mb-3` to space it out from the FormSwitchGroup below.  The `mt-4` here is something that I'd probably add to the parent component/previous component in the parent.

davidjoy reviewed (approved):
Approving, though there's a conflict that'll need to be addressed.  Note that a recent PR changed the data model and API layer a bit, I'd do a quick search for `allowDivisionByUnit` after rebasing to make sure all references to it are gone.

I also noted one small nit/stylistic change that I'd fix to primarily use bottom margins instead of top margins; more details in comments.

awaisdar001 commented:
Please rebase before merging.
